Sources/FoundationEssentials/String/String+Path.swift

```diff
@@ -137,6 +137,24 @@ extension String {
     
     #if !NO_FILESYSTEM
     internal static func homeDirectoryPath(forUser user: String? = nil) -> String {
+#if os(Windows)
+        var hToken: HANDLE? = nil
+        guard OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &hToken) else {
+            return ""
+        }
+        defer { CloseHandle(hToken) }
+
+        var dwcchSize: DWORD = 0
+        _ = GetUserProfileDirectoryW(hToken, nil, &dwcchSize)
```

Comment:
It looks like we're ignoring the provided `user` value here - do we want to support that? I see that the CF source within swift-corelibs-Foundation on windows has a length C implementation to lookup another user's home directory at https://github.com/apple/swift-corelibs-foundation/blob/77af9c5a984f40e186c0ba127cff88f2a2588c15/CoreFoundation/Base.subproj/CFPlatform.c#L619, maybe we should port that to Swift here?

Reply:
Oh, yeah, if we need to be able to query another user's home directory, that would need to be done. I forgot that I had written that code in C 😂

Reply:
Yeah it's not super common (it's really used to support the `FileManager.homeDirectory(forUser:)` API + standardizing paths that look like `~userName/foo`) but something we'll want to support here in addition to the current user (when the parameter is `nil`).

Haha yeah I assumed it was probably you who wrote that in C back in the day, hopefully it can be ported to Swift without too much extra leg work

Reply:
It is patently ugly, but I think it should work?